content/post/087-js-ipfs-0.42.md

```diff
@@ -0,0 +1,194 @@
+---
+date: 2020-04-14
+url: 2020-04-14-js-ipfs-0-42
+title: js-ipfs 0.42.0 released
+author: Alex Potsides
+---
+
+# 🔦 Highlights
+
+> Smaller, faster, more reliable
+
+`js-IPFS@0.42.0` has been released, bringing a host of bug fixes and other small stability improvements.
+
+It reduces the size of the minified bundle by 7.5% for faster downloads and more responsive web apps!
+
+## 🤖 Automated publishing of RCs
+
+Release Candidate builds of [`js-ipfs`](https://www.npmjs.com/package/ipfs) and [`ipfs-http-client`](https://www.npmjs.com/package/ipfs-http-client) are now being published to npm with every build of master, so you can try out the very bleeding edge with:
+
+```console
+$ npm install -g ipfs@next
+or
+$ npm install ipfs-http-client@next
+```
+
+This will download and install a version of `js-ipfs` and/or `ipfs-http-client` created from the last successful [build](https://travis-ci.com/github/ipfs/js-ipfs/branches) of master with no github URLs in the production dependency tree.
+
+## 🛳️ Updated Docker containers
+
+Publishing to Docker Hub has been restored so you can get a container running the latest and greatest `js-ipfs` with:
+
+```console
+$ docker pull ipfs/js-ipfs
+```
+
+You can also pull pre-releases at any time with:
+
+```console
+$ docker pull ipfs/js-ipfs:next
+```
+
+Check out all the releases at: https://hub.docker.com/r/ipfs/js-ipfs
+
+## ✨New features
+
+The [MFS][] CLI now supports nanosecond mtimes, use it with the following commands:
+
+```console
+$ jsipfs mkdir /directory --mtime 1000 --mtime-nsecs 5
+```
+
+`--mode` and `--mtime` are now also supported when piping files to `jsipfs add`:
+
+```console
+$ cat ./file.txt | jsipfs add --mode 0500
+```
+
+## 🔨 Breaking changes
+
+- [HAMT](https://en.wikipedia.org/wiki/Hash_array_mapped_trie) sharded directories are now reported as having a type of `'directory'` from `ipfs.files.stat` in line with `go-ipfs`
+- When specifying metadata and `--raw-leaves` to `ipfs.add`, if the file is small enough to fit into one block, `--raw-leaves` is ignored and a [UnixFS][] entry is created (previously the returned [CID][] would have a codec of `raw`)
+
+## 🦟 Bugs fixed
+
+- WebRTC configuration is now passed to the libp2p WebRTC transport correctly
+- Unknown CLI commands now cause the process to exit with a non-zero exit code
+- Running Prometheus metrics in Docker containers no longer prevent the container from starting
+
+## 💍 One PR to rule them all
```

Comment:
"Lerna Monorepo" might be a more attention grabbing and exciting headline for this.

Reply:
I'm all up for more exciting headlines, it just makes me groan a bit - it's not really a monorepo because there are still plenty of modules used pretty much exclusively by js-ipfs in other repos and lerna is just a tool, I really wanted to concentrate on the why not the how.